From 179b1b64275757253932c5663d058017d060a67f Mon Sep 17 00:00:00 2001
From: Li Nan <linan122@huawei.com>
Date: Tue, 6 Sep 2022 10:01:28 +0800
Subject: [PATCH 245/261] [Huawei] Revert "[Backport] block/wbt: fix negative
 inflight counter when remove scsi device"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187584

--------------------------------

This reverts commit 60e9211aed188e2269d56c4a08fa6807adb20781.

There are two wbt_enable_default() in bfq_exit_queue(), and it will reset
inflight. When io is issued between two enable, inflight might becomce -1.
Revert one.

Signed-off-by: Li Nan <linan122@huawei.com>
---
 block/bfq-iosched.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/block/bfq-iosched.c b/block/bfq-iosched.c
index fd34bc70b46a..fa06f8069274 100644
--- a/block/bfq-iosched.c
+++ b/block/bfq-iosched.c
@@ -6418,8 +6418,6 @@ static void bfq_exit_queue(struct elevator_queue *e)
 	spin_unlock_irq(&bfqd->lock);
 #endif
 
-	wbt_enable_default(bfqd->queue);
-
 	kfree(bfqd);
 
 	/* Re-enable throttling in case elevator disabled it */
-- 
2.17.1

